Source/WebCore/accessibility/AXObjectCache.cpp

```diff
@@ -1300,46 +1300,17 @@ void AXObjectCache::handleTextChanged(AccessibilityObject* object)
     object->recomputeIsIgnored();
 }
 
-void AXObjectCache::onRendererCreated(Element& element)
-{
-    if (!element.renderer()) {
-        ASSERT_NOT_REACHED();
-        return;
-    }
-
-    // If there is already an AXObject that was created for this element,
-    // remove it since there will be a new AXRenderObject created using the renderer.
-    if (auto axID = m_nodeObjectMapping.get(element)) {
-        // The removal needs to be async because this is called during a RenderTree
-        // update and remove(AXID) updates the isolated tree, that in turn calls
-        // parentObjectUnignored() on the object being removed, that may result
-        // in a call to textUnderElement, that can not be called during a layout.
-        m_deferredReplacedObjects.add(*axID);
-        if (!m_performCacheUpdateTimer.isActive())
-            m_performCacheUpdateTimer.startOneShot(0_s);
-    }
-}
-
-void AXObjectCache::onRendererCreated(Text& textNode)
+void AXObjectCache::onRendererCreated(Node& node)
 {
-    if (!textNode.renderer()) {
+    CheckedPtr renderer = node.renderer();
+    if (!renderer) [[unlikely]] {
         ASSERT_NOT_REACHED();
         return;
     }
 
-    // If we created an AccessibilityNodeObject for this Text, remove it since there should
-    // be a new AccessibilityRenderObject created using the renderer.
-    if (auto axID = m_nodeObjectMapping.get(textNode)) {
-        if (RefPtr nodeObject = get(textNode)) {
-            if (auto* parent = nodeObject->parentObject()) {
-                remove(textNode);
-                childrenChanged(parent);
-                return;
-            }
-            // We don't need to add nodeObject to m_deferredReplacedObjects, as that currently
-            // only serves to repair relationships for replaced objects, which text nodes cannot
-            // possibly be part of (because they are not elements).
-        }
+    if (RefPtr existingObject = dynamicDowncast<AccessibilityRenderObject>(get(node))) {
```

Comment:
It looks like this handles the case where we already have an object associated with the node. What if we don't know about this object yet? Is this a safe time to call getOrCreate(node)? Are we just assuming we'd get a different callback about the node?